### ISSUE TITLE
Unable to connect to OneView if private CA is used and CRL is not available

### ISSUE THREAD

**Issue by martin-rumpel** [HewlettPackard/POSH-HPOneView#97]
If tired connect to OneView via  Connect-HPOVMgmt  and HP OneView use certificate issued by externall private CA, but
CRL is not available (because of system out of internet), connection failed because of unable to verify CRL.

As workaroud i change to self-signed certificate for OneView, which works good.

### Version Information
HPE OneView PowerShell Library Version (`Get-HPOVVersion`): 2.0.300.0
HPE OneView Appliance Version:  2.00.02-0219211
Output from `$PSVersionTable` on your Windows Host: 
Name  : PSVersion
Value : 4.0

Name  : WSManStackVersion
Value : 3.0

Name  : SerializationVersion
Value : 1.1.0.1

Name  : CLRVersion
Value : 4.0.30319.42000

Name  : BuildVersion
Value : 6.3.9600.17400

Name  : PSCompatibleVersions
Value : {1.0, 2.0, 3.0, 4.0}

Name  : PSRemotingProtocolVersion
Value : 2.2

Thanks a lot.

Martin

**Comment by ChrisLynchHPE** [comment]
Can you provide a screenshot or capture the output from the `Connect-HPOVMgmt` call?  I want to see what the certificate status is.  Also, are you connecting to an appliance with either the IP Address or FQDN, and is it in either the Subject or Subject Alternative Name (SAN) value?

**Comment by martin-rumpel** [comment]
I saved output into attached text file. I also attach certificate. I use FQDN, but with IP also won't work...

[output.txt](https://github.com/HewlettPackard/POSH-HPOneView/files/225703/output.txt)
[certificates.zip](https://github.com/HewlettPackard/POSH-HPOneView/files/225704/certificates.zip)

**Comment by martin-rumpel** [comment]
I saved output into attached text file. I also attach certificate. I use FQDN, but with IP also won't work...

**Comment by ChrisLynchHPE** [comment]
Thanks.  I will look at this in greater detail to see what is going on.

**Comment by ChrisLynchHPE** [comment]
So, does your system Trust the Issuing CA?  I wonder if the `Import-HPOVSSLCertificate` didn't import the cert into your PC's trusted store.  Can you provide the output from:

```PowerShell
Get-ChildItem -path Cert:\CurrentUser\Root | ? ThumbPrint -eq 659F036B97D774341EF1C22A14C6F78B2FAE05CC | FL *
```

**Comment by martin-rumpel** [comment]
Root certificate is distributed by GPO into local computer\Trused Root CA
Also if I use IE and go to webpage of HPOneView certificate seems good.

I suppose that possibly problem with CA CRL. My infrastructure is out of internet and out of my CA, therefore no CLR record is in certificate.

After `Connect-HPOVMgmt` validation of certificate fails with:
`CertificateIsValid      : False
ErrorInformation        : **RevocationStatusUnknown**`

It seems that validation tries to chack revocation status... And if can't check it, validation fails... 


Result of cet instalation:
`PS C:\Windows\system32> Get-ChildItem -path Cert:\CurrentUser\Root | ? ThumbPrint -eq 659F036B97D774341EF1C22A14C6F78B2FAE05CC


    Directory: Microsoft.PowerShell.Security\Certificate::CurrentUser\Root


Thumbprint                                Subject
----------                                -------
659F036B97D774341EF1C22A14C6F78B2FAE05CC  CN=hponeview.aszd.local, OU=ASZD, O=SEPS, L=Bratislava, S=Slovakia, C=SK`

**Comment by martin-rumpel** [comment]
I tried change CA, before CRL Distribution point list was empty, now I add one CLR distribution point ( http://Service/CertEnroll/DataSystemSoftServiceCA.crl ) into cert. Of corse this point if not reachable from my infrastructure.

Error information during validation of certificate changed

after: `ErrorInformation : RevocationStatusUnknown `
before `ErrorInformation : RevocationStatusUnknown OfflineRevocation`

But still ends on error because of unable to verify CRL...

**Comment by ChrisLynchHPE** [comment]
> Is any possibility how to disable CRL check?
That kind of defeats the purpose of Certificate Authorities.  CRL is provided to make sure that the cert is still valid and hasn't been revoked.

I think I have this fixed.  I have added support to explicitly trust the cert if the `ErrorInformation` of the cert contains either `RevocationStatusUnknown` or `OfflineRevocation`.  I don't have an environment to test this with.  So I will publish an update, and you can either reopen this issue or create a new one.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 2.0.368.0](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v2.0.368.0).

Closing.
